library/common/http/client.cc

```diff
@@ -471,9 +471,11 @@ void Client::sendData(envoy_stream_t stream, envoy_data data, bool end_stream) {
     if (direct_stream->explicit_flow_control_ && !end_stream) {
       if (direct_stream->read_disable_count_ == 0) {
         // If there is still buffer space after the write, notify the sender
-        // that send window is available.
+        // that send window is available, on the next dispatcher iteration.
         direct_stream->wants_write_notification_ = false;
-        direct_stream->callbacks_->onSendWindowAvailable();
+        scheduled_callback_ = dispatcher_.createSchedulableCallback(
```

Comment:
can't remember offhand: do we need to create a new one each time, or only the one time?

Reply:
Oh nice. That seems to work. Done!

Reply:
apparently it didn't work - can we call out why we always need to create in a comment?

Reply:
Good point. Done.